### PR TITLE
[2.4] Check wandb login

### PR DESCRIPTION
**Pull request by SYangster**
- Fixes [1890](https://jirasw.nvidia.com/browse/FLARE-1890)
- Check wandb login, run job to completion without wandb logging if login is unsuccessful 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_opt/tracking/wandb/wandb_receiver.py [hunk 1]
```diff
@@ -82,6 +82,13 @@ def initialize(self, fl_ctx: FLContext):
         run_name = self.kwargs["name"]
         job_id_tag = self.get_job_id_tag(run_group_id)
         wand_config = self.kwargs.get("config", {})
+
+        try:
+            wandb.login(timeout=1)
+        except Exception as e:
+            self.log_error(self.fl_ctx, f"Unsuccessful login: {e}. Will not log to wandb.")
+            return
+
```
Comment (YuanTingHsieh) [inline nvflare/app_opt/tracking/wandb/wandb_receiver.py:91]: Have you tested this on both "online" and "offline" mode?

I wonder how do we pass server URL or something to this WandBReceiver?

Comment (SYangster) [reply nvflare/app_opt/tracking/wandb/wandb_receiver.py]: Changed to use offline mode as a default rather than not logging at all.

Comment (YuanTingHsieh) [reply nvflare/app_opt/tracking/wandb/wandb_receiver.py]: thanks, for this question "how do we pass server URL or something to this WandBReceiver?" we can leave it to future PR.

**Review by YuanTingHsieh — DISMISSED** [review]

File: nvflare/app_opt/tracking/wandb/wandb_receiver.py [hunk 2]
```diff
@@ -82,10 +82,18 @@ def initialize(self, fl_ctx: FLContext):
         run_name = self.kwargs["name"]
         job_id_tag = self.get_job_id_tag(run_group_id)
         wand_config = self.kwargs.get("config", {})
+
+        try:
+            wandb.login(timeout=1, verify=True)
+        except Exception as e:
+            self.log_error(self.fl_ctx, f"Unsuccessful login: {e}. Using wandb offline mode.")
+            self.mode = "offline"
```
Comment (YuanTingHsieh) [inline nvflare/app_opt/tracking/wandb/wandb_receiver.py:90]: actually can you add an if statement here?
`if self.mode == "online"`

File: nvflare/app_opt/tracking/wandb/wandb_receiver.py [hunk 3]
```diff
@@ -82,10 +82,19 @@ def initialize(self, fl_ctx: FLContext):
         run_name = self.kwargs["name"]
         job_id_tag = self.get_job_id_tag(run_group_id)
         wand_config = self.kwargs.get("config", {})
+
+        try:
+            if self.mode == "online":
+                wandb.login(timeout=1, verify=True)
+        except Exception as e:
+            self.log_error(self.fl_ctx, f"Unsuccessful login: {e}. Using wandb offline mode.")
+            self.mode = "offline"
```
Comment (YuanTingHsieh) [inline nvflare/app_opt/tracking/wandb/wandb_receiver.py:91]: I prefer this, but minor:
```
    if self.mode == "online":
        try:
            wandb.login(timeout=1, verify=True)
        except Exception as e:
            self.log_error(self.fl_ctx, f"Unsuccessful login: {e}. Using wandb offline mode.")
            self.mode = "offline"
```

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM

**Pull request merged**